frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -24,13 +26,15 @@ end
 
 function filemanagerutil.abbreviate(path)
     if not path then return "" end
-    local home_dir_name = G_reader_settings:readSetting("home_dir_display_name")
-    if home_dir_name ~= nil then
+    if G_reader_settings:isTrue("shorten_home_dir") then
         local home_dir = G_reader_settings:readSetting("home_dir") or filemanagerutil.getDefaultDir()
+        if path == home_dir then
+            return home_dir:sub(m_util.lastIndexOf(home_dir, "/")+1)
+        end
         local len = home_dir:len()
         local start = path:sub(1, len)
         if start == home_dir then
-            return home_dir_name .. path:sub(len+1)
+            return path:sub(len+1)
```

Comment:
You didn't need the `len+2` from my sample code to remove the leading `/`
(We need to remove it, to distinguish shortened paths inside home_dir from full path outside home_dir.)

Reply:
Oops, missed this earlier, implemented now